examples/alerts/alerts.yaml

```
@@ -321,14 +311,18 @@ groups:
     for: 5m
     labels:
       severity: critical
-  - alert: ThanosSidecarUnhealthy
+  - alert: ThanosSidecarNoConnectionToStartedPrometheus
     annotations:
       description: Thanos Sidecar {{$labels.instance}} is unhealthy for more than
         {{$value}} seconds.
-      runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanossidecarunhealthy
-      summary: Thanos Sidecar is unhealthy.
+      runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanossidecarnoconnectiontostartedprometheus
+      summary: Thanos Sidecar cannot access Prometheus, even though Prometheus seems
+        healthy and has reloaded WAL.
     expr: |
-      time() - max by (job, instance) (thanos_sidecar_last_heartbeat_success_time_seconds{job=~".*thanos-sidecar.*"}) >= 240
+      time() - max by (pod, job, instance) (thanos_sidecar_last_heartbeat_success_time_seconds{job=~".*thanos-sidecar.*"}) >= 240
```

Sorry for challenging this last minute, but isn't expression with `thanos_sidecar_prometheus_up` simpler?

 ```suggestion
      thanos_sidecar_prometheus_up{job=~".*thanos-sidecar.*"}) == 0
```

---

Great question :) AFAIK, thanos_sidecar_last_heartbeat_success_time_seconds and thanos_sidecar_prometheus_up carries the same information encoded in different ways. Let me evaluate and switch to  thanos_sidecar_prometheus_up if it is the case.

---

@bwplotka If we switch to thanos_sidecar_last_heartbeat_success_time_seconds, We may not able to show the duration at which sidecar is not connected in the alert description. Hope that is fine.
e.g.
> Thanos Sidecar thanos-sidecar-1 is unhealthy for more than 720 seconds.